app/components/confirmation-modal/component.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+import layout from './template';
+
+export default Ember.Component.extend({
+  confirmationModalService: Ember.inject.service('confirmation-modal'),
+  layout: layout,
+
+  model: Ember.computed.reads('modal.model'),
+
+  init() {
+    this._super.apply(this, arguments);
+    this.get('confirmationModalService').on('open', (modal) => {
+      this.set('modal', modal);
+      this.set('isOpen', true);
+    });
+  },
```

Comment:
So, the future of `on('init')` type callbacks seems unclear. I don't think there is much interest in removing them.

There might be some curse of expertise going on here, I definitely have a gut feeling that I want to use `init() {` instead of an evented hook. Not sure why.